### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by jishnub**
Initial version of docs, with some text from the README, a list of docstrings using `@autodocs`, and some text added by me in a quick-start section.

cc @ChrisRackauckas

### CODE REVIEW

**Review by ChrisRackauckas — APPROVED** [review]
Thanks! This looks great.

File: docs/src/index.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+2×5 Fill{Float64}, with entries equal to 2.0
+
+julia> println.(Fill(pi, 10))
+π
+10-element Fill{Nothing}, with entries equal to nothing
+```
+
+Notice that this will only match the behaviour of a dense matrix from `fill` if the function is pure. And that this shortcut is taken before any other fused broadcast:
+
+```julia
```
Comment (dlfivefifty) [inline docs/src/index.md:153]: Set `Random.seed!` so we can jldoctest these?

**Review by dlfivefifty — CHANGES_REQUESTED** [review]

File: docs/src/index.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
+```@meta
+DocTestSetup  = quote
+    using FillArrays
+end
+```
+
+# Introduction
+
+`FillArrays` allows one to lazily represent arrays filled with a single entry, as well as identity matrices. This package exports the following types: `Eye`, `Fill`, `Ones`, `Zeros`, `Trues` and `Falses`.
```
[the remark this answers is not in the document]
Comment (jishnub) [reply docs/src/index.md]: Updated now

File: docs/src/index.md (hunk 1 above)
Comment (jishnub) [reply docs/src/index.md]: Updated

**Comment by jishnub** [comment]
Gentle bump

**Pull request merged**